lib/blocklist.c

```diff
@@ -102,8 +102,33 @@ void allowlist_prefix(char *ip, int prefix_len)
 	_add_constraint(addr, prefix_len, ADDR_ALLOWED);
 }
 
+static int is_ip_ipv6(char *ip) {
+	// get length of string including the null-byte
+	int length = strlen(ip) + 1;
+	// don't modify the input string
+	char *new_str = strndup(ip, length);
```

Comment:
I don't really understand the intent behind calling `strlen` and then passing that into `strndup`. This feels equivalent to using `strdup`. If you want to have a limited length, we should use `strndup` and check that the length is below a certain size using `strnlen`. Otherwise, we might as well just use `strdup`.

Reply:
Yeah, not sure what I was thinking, `strdup` looks ideal for this.